angular-client/src/components/carousel/carousel.component.ts

```diff
@@ -26,16 +28,36 @@ export class Carousel {
     this.runs = data.runs;
   }
 
+  ngOnInit(): void {
+    this.currentIndex = 0;
+  }
+
+  updateIndex(nexIndex: number) {
+    this.currentIndex = nexIndex;
+  }
+
   onNoClick(): void {
     this.dialogRef.close();
   }
 
+  handlePageChange(event: any): void {
+    // Assuming event.page gives the new index
+    const newIndex = event.page;
+
+    if (newIndex > this.previousIndex) {
+      this.currentIndex = Math.min(this.currentIndex + 1, this.runs.length - 1);
```

Comment:
your math here is corrupted if you go over the runs.length - 1 you should set your index to 0